### PR TITLE
fix(yaml): do not throw errors for multi doc yaml

### PR DESCRIPTION
**Pull request by Klaus-Keller**
Issue https://github.com/SAP/open-ux-tools/issues/892

## Description
This PR fixes errors thrown in module `@sap-ux/yaml` when parsing multi document yaml strings. While it prepares for working with multiple documents, it does not provide full multi document capabilities. Full support of multi document yaml requires more technical evaluation and planing.

## Background
When running 
https://github.com/SAP/open-ux-tools/blob/41e234c0ba70425b42139a42218ece5a3562c032/packages/yaml/src/yaml-document.ts#L41-L47

with a `serializedYaml` containing [mulitple documents](https://www.yaml.info/learn/document.html), `this.document.error` contains at least one error with `code: 'MULTIPLE_DOCS'`. However, the document would still be parsed. Thus another possible fix for this would be to add an error filter to the constructor, like
`this.document.errors = this.document.errors?.filter((error) => error.code !== 'MULTIPLE_DOCS');`

Resulting constructor looks then like this:

```typescript
private constructor(serializedYaml: string) {
    this.document = yaml.parseDocument(serializedYaml);
    this.document.errors = this.document.errors?.filter((error) => error.code !== 'MULTIPLE_DOCS');
    if (this.document.errors?.length > 0) {
        throw new YAMLError(
            errorTemplate.yamlParsing + '\n' + this.document.errors.map((e) => e.message).join(''),
            errorCode.yamlParsing
        );
    }
}
```

However, this solution would have a flaw: 
When providing a yaml with multiple documents and calling `doc.toString()` at some point, only the first document is serialized.

The proposed solution on this PR also prepares for full support of multiple document yamls. For instance, the index of the document (currently fixed to `0`) could be optional parameter for function calls allowing to reference a specific document in the yaml.

### CODE REVIEW

File: packages/yaml/src/yaml-document.ts [hunk 1]
```diff
@@ -38,10 +38,16 @@ export class YamlDocument {
...
-        this.document = yaml.parseDocument(serializedYaml);
-        if (this.document.errors?.length > 0) {
+        this.documents = yaml.parseAllDocuments(serializedYaml);
+        if (this.documents.length === 0) {
+            // this is done for backward compatibility when only single document yaml was supported.
+            // yaml.parseDocument('') creates a new document
+            // yaml.parseAll('') does not create a new document
+            this.documents.push(yaml.parseDocument(serializedYaml));
+        }
+        if (this.documents[0].errors?.length > 0) {
```
Comment (ullasholla) [inline packages/yaml/src/yaml-document.ts:48]: Consider errors in the other documents as well?

**Review by ullasholla — CHANGES_REQUESTED** [review]
Thanks @Klaus-Keller, the code looks good. The test coverage is very good. Had one question about error handling

File: packages/yaml/src/yaml-document.ts (hunk 1 above)
Comment (Klaus-Keller) [reply packages/yaml/src/yaml-document.ts]: added it with https://github.com/SAP/open-ux-tools/pull/906/commits/ce25fc781ab51ad569d1dfd679865a49d4d199c3

**Review by ullasholla — APPROVED** [review]

**Review by tobiasqueck — APPROVED** [review]
* code adheres to coding standards
* functionality is as expected
* tests properly extended for the new case

**Pull request merged**